text/0000-plan-flags.md

```diff
@@ -9,206 +9,238 @@
+  [requires.capabilities]
+    build = false
+    launch = false
+  [requires.modifiers]
+    some-modifer = "modifier"
   
   [[or]]
   
   [[or.provides]]
   name = "<dependency name>"
+  strategy = "<merging strategy>"
   
   [[or.requires]]
   name = "<dependency name>"
-  version = "<other dependency version>"
-  build = true
-  launch = false
-  
-  [or.requires.metadata]
-  some_other_metadata_key = "some_other_metadata_value"
-  # buildpack-specific data
-  
+  [requires.version]
+    constraint = "<other dependency constraint>"
+    source = "<other dependency constraint source>"
+  [requires.capabilities]
+    build = false
+    launch = false
+  [requires.modifiers]
+    some-other-modifer = "other modifier"
   ```
+
+This proposal removes both the `version` and `metadata` fields and replaces them with three new fields each with their own purpose.
+The `requires.version` field holds all of the version requirement information, this includes a `constraint` which can be any semver constraint
+to be used when deciding what dependency the providing buildpack should install and the `source` so that the origin of the constraint can be identified
+this will be important when discussing the merging done by the lifecycle later. `requires.capablities` hold the `build` and `launch` keys and indicate
+when during which phase buildpack specific data and dependencies are available. Finally, there is `requires.modifiers` this is meant to hold only specific
+and unique buildpack communications.
   
 ## Buildpack Plan (TOML) changes
   
-  For the same reasons, we would like to add `build` and `launch` flags to the Buildpack Plan, along with
+  For the same reasons, we would like to add the `version`, `capabilities`, and `modifiers` fields to the Buildpack Plan, along with a `versions` field and
   some additional format changes: 
 
   ```toml
   [[entries]]
   name = "<dependency name>"
-  build = false
-  launch = false
-  
-  [[entries.requires]]
-  version = "<dependency version>"
-  
-  [entries.requires.metadata]
-  # buildpack-specific data
+  [entries.version]
+    constraint = "<dependency constraint>"
+  [entries.capabilities]
+    build = false
+    launch = false
+  [entries.modifiers]
+    # arbitrary keys and values
+  [[entries.versions]]
+    constraint = “<dependency constraint>”
+    source = "<dependency constraint source>"
   ```
 In each entry the `name` field is **unique**.
 
   We arrived at this format for several reasons:
   
-  - It allows us to offload some common flag merging operations to the lifecycle. We suggest letting all build/launch flags
-  under a single key be merged together (recommending the `or` operator). 
+  - It allows us to offload some common field merging operations to the lifecycle. We suggest that `version` is merged together if the merge would result in
+  a valid semver constraint. If the merge fails to produce a valid semver constraint it would cause the image build to fail.
```

Comment:
How semver merging would work for the following?
- `> 1.5.0`, `< 1.6.0`
Do constrains have to be nested?

Reply:
I think that this is an implementation detail that should be resolved at the time of implementation but I assume that it would result in an `and` semver constraint.